shared/src/main/scala/mlscript/codegen/Symbol.scala

```diff
@@ -0,0 +1,158 @@
+package mlscript.codegen
+
+import mlscript.utils.shorthands._
+import mlscript.Type
+import mlscript.JSClassDecl
+import mlscript.MethodDef
+import mlscript.Term
+import mlscript.TypeName
+
+sealed trait LexicalSymbol {
+
+  /**
+    * The lexical name of the symbol. This is different from the runtime name,
+    * the name of the symbol in the generated code. We allow duplicates lexical
+    * names in the same scope.
+    */
+  def lexicalName: Str
+}
+
+sealed trait RuntimeSymbol extends LexicalSymbol {
+  def runtimeName: Str
+}
+
+sealed trait TypeSymbol extends LexicalSymbol {
+  val params: Ls[Str]
+  val actualType: Type
+}
+
+sealed class ValueSymbol(val lexicalName: Str, val runtimeName: Str) extends RuntimeSymbol {
+  override def toString: Str = s"value $lexicalName"
+}
+
+object ValueSymbol {
+  def apply(lexicalName: Str, runtimeName: Str): ValueSymbol =
+    new ValueSymbol(lexicalName, runtimeName)
+}
+
+sealed case class TypeAliasSymbol(
+    val lexicalName: Str,
+    val params: Ls[Str],
+    val actualType: Type
+) extends TypeSymbol
+    with LexicalSymbol {
+  override def toString: Str = s"type $lexicalName"
+}
+
+final case class BuiltinSymbol(val lexicalName: Str, feature: Str) extends RuntimeSymbol {
+  val runtimeName = lexicalName
+
+  override def toString: Str = s"function $lexicalName"
+
+  /**
+    * `true` if the built-in value had been accessed before.
+    * `Scope` will reuse the `lexicalName` if `accessed` is false.
+    */
+  var accessed: Bool = false
+}
+
+final case class StubValueSymbol(
+    override val lexicalName: Str,
+    override val runtimeName: Str,
+    /**
+      * Whether this stub is shadowing another symbol.
+      */
+    val shadowing: Bool,
+    previous: Opt[StubValueSymbol]
+)(implicit val accessible: Bool)
+    extends RuntimeSymbol {
+  override def toString: Str = s"value $lexicalName"
+}
+
+final case class ClassSymbol(
+    lexicalName: Str,
+    runtimeName: Str,
+    params: Ls[Str],
+    actualType: Type,
```

Comment:
Why is the body type called the "actual" type? Very strange and nonstandard terminology. Please change.

Reply:
It was because there had been another member named `body`. Changed in `52324df`.